features/fields-generator/fields_mapping_generator.py

```diff
@@ -0,0 +1,218 @@
+info = False
+settingsFileName = ""
+
+for argument in sys.argv:
+    if (argument[:8]=="--config"):
+        settingsFileName = argument[8:]
+    if (argument=="--debug" or argument=="-d"):
+        debug = True
+        info = True
+    elif (argument=="--info" or argument=="-i"):
+        info = True
+    elif (argument=="--help" or argument=="-h"):
+        print ('Command line arguments:')
+        print ('--config         bestand van projectspecifieke configuratie json bestand')
+        print (' --info    -i    toon voortgangs info')
+        print (' --debug   -d    toon debug info')
+        print (' --help    -h    toon help op command line arguments')
+        sys.exit()
+
+# lees configuratiebestand
+if settingsFileName!="":
+    with open(settingsFileName) as settingsFile:
+        SETTINGS = json.load(settingsFile)
+else:
+    print ("Geef het gewenste config bestand op met parameter --config.")
+    sys.exit()
+
+if (not os.path.isdir(SETTINGS.get("projectFolder"))):
+    print ("Geef een correcte projectFolder in het config bestand.")
+    sys.exit()
+
+if (not os.path.isfile(SETTINGS.get("projectFolder") + SETTINGS.get("sourceYaml"))):
+    print ("Geef een correcte locatie op voor de API specificaties in sourceYaml.")
+    sys.exit()
+
+if (not os.path.isdir(SETTINGS.get("projectFolder") + SETTINGS.get("sheetFolder"))):
+    print ("Geef een correcte locatie op voor de te genereren sheets in sheetFolder.")
+    print (SETTINGS.get("projectFolder") + SETTINGS.get("sheetFolder"))
+    sys.exit()
+
+
+#lees bron API specificatiesbestand naar een Dict
+SWAGGER = yaml.full_load(open(SETTINGS.get("projectFolder") + SETTINGS.get("sourceYaml")))
+
+# voor elk schemaComponents item in het configuratiebestand wordt een bestand gegenereerd
+for schemaComponent in SETTINGS.get("schemaComponents"):
+    if info==True:
+            print (schemaComponent.get("name"))
+
+    filePath = SETTINGS.get("projectFolder") + SETTINGS.get("fieldslistFolder") + "fields-mapping-"  + schemaComponent.get("name") + ".csv"
+
+    # lees alle velden uit de API specificaties van de resource schemacomponent
+    fields = []
+    writeComponent ("", "#/components/schemas/" + schemaComponent.get("name"))
+    
+    # verwijder dubbele fields paden
+    fields = list(dict.fromkeys(fields)) 
+
+    fieldPaths = []
+    for field in fields:
+        fieldPaths.append({ "shortPath": field, "fullPath": field})
+
+        # voeg verkort pad toe aan veld
+        pathList = field.split(".")
+        if len(pathList)>1:
+            for i in range(-len(pathList)+1, 0):
+                fieldPaths.append({ "shortPath": ".".join(pathList[i:]), "fullPath": field})
+
+    # verwijder niet-unieke paden
+    unfilteredFieldPaths = fieldPaths
+    fieldPaths = filter(unique, fieldPaths)
+
+    # verwijder deelvelden van een datum
+    fieldsList = list(map(removeDateField, fieldPaths))
+
+    # voeg gba mapping toe
+    fieldsList = list(map(addGbaMapping, fieldsList)) 
```

Comment:
addGbaMapping komt niet voor als functie

Reply:
grrrr, te lang geleden dat ik het script gedraaid had. stap is niet nodig, verwijder ik